File: mobile/src/main/java/treehou/se/habit/util/logging/FirebaseLogger.java
package treehou.se.habit.util.logging;

import android.util.Log;

import com.google.firebase.crash.FirebaseCrash;

public class FirebaseLogger implements Logger {

    @Override
    public void d(String tag, String message) {
        Log.d(tag, message);
    }

    @Override
    public void i(String tag, String message) {
        Log.i(tag, message);
    }

    @Override
    public void w(String tag, String message) {
        Log.w(tag, message);
    }

    @Override
    public void w(String tag, String message, Throwable error) {
        Log.e(tag, message, error);
        FirebaseCrash.report(error);
    }

    @Override
    public void e(String tag, String message, Throwable error) {
        Log.e(tag, message, error);
        FirebaseCrash.report(error);
    }

    @Override
    public void e(String tag, String message) {
        Log.e(tag, message);
    }
}

File: mobile/src/main/java/treehou/se/habit/util/logging/FirebaseLogger.kt
package treehou.se.habit.util.logging

import android.util.Log

import com.google.firebase.crash.FirebaseCrash

class FirebaseLogger : Logger {

    override fun d(tag: String, message: String) {
        Log.d(tag, message)
    }

    override fun i(tag: String, message: String) {
        Log.i(tag, message)
    }

    override fun w(tag: String, message: String) {
        Log.w(tag, message)
    }

    override fun w(tag: String, message: String, error: Throwable) {
        Log.e(tag, message, error)
        FirebaseCrash.report(error)
    }

    override fun e(tag: String, message: String, error: Throwable) {
        Log.e(tag, message, error)
        FirebaseCrash.report(error)
    }

    override fun e(tag: String, message: String) {
        Log.e(tag, message)
    }
}
